src/Liquidator.sol

```diff
@@ -214,267 +224,304 @@ contract Liquidator is Owned {
-     * MPM: The minPriceMultiplier defines the asymptotic end price for P(∞) = openDebt * MPM (2 decimals precision).
-     * base: defines how fast the exponential curve decreases (18 decimals precision).
-     * t: time passed since start auction (in seconds, 18 decimals precision).
-     * @dev LogExpMath was made in solidity 0.7, where operations were unchecked.
+     * @notice Calculate asset distribution percentages based on provided risk values.
+     * @param riskValues_ An array of risk values for assets.
+     * @return assetDistributions An array of asset distribution percentages (in tenths of a percent, e.g., 1_000_000 represents 100%).
      */
-    function _calcPriceOfAccount(AuctionInformation memory auctionInfo) internal view returns (uint256 price) {
-        //Time passed is a difference of two Uint32 -> can't overflow.
-        uint256 timePassed;
-        unchecked {
-            timePassed = block.timestamp - auctionInfo.startTime; //time duration in seconds.
-
-            if (timePassed > auctionInfo.cutoffTime) {
-                //Cut-off time passed -> return the minimal value defined by minPriceMultiplier (2 decimals precision).
-                //No overflow possible: uint128 * uint8.
-                price = uint256(auctionInfo.openDebt) * auctionInfo.minPriceMultiplier / 1e2;
-            } else {
-                //Bring to 18 decimals precision for LogExpMath.pow()
-                //No overflow possible: uin32 * uint64.
-                timePassed = timePassed * 1e18;
-
-                //pow(base, timePassed) has 18 decimals and is strictly smaller than 1 (-> smaller as 1e18).
-                //No overflow possible: uint128 * uint64 * uint8.
-                //Multipliers have 2 decimals precision and LogExpMath.pow() has 18 decimals precision,
-                //hence we need to divide the result by 1e20.
-                price = auctionInfo.openDebt
-                    * (
-                        LogExpMath.pow(auctionInfo.base, timePassed)
-                            * (auctionInfo.startPriceMultiplier - auctionInfo.minPriceMultiplier)
-                            + 1e18 * uint256(auctionInfo.minPriceMultiplier)
-                    ) / 1e20;
+    function _getAssetDistribution(RiskModule.AssetValueAndRiskFactors[] memory riskValues_)
+        internal
+        pure
+        returns (uint32[] memory assetDistributions)
+    {
+        uint256 length = riskValues_.length;
+        uint256 totalValue;
+        for (uint256 i; i < length;) {
+            unchecked {
+                totalValue += riskValues_[i].assetValue;
+            }
+            unchecked {
+                ++i;
+            }
+        }
+        assetDistributions = new uint32[](length);
+        for (uint256 i; i < length;) {
+            // The asset distribution is calculated as a percentage of the total value of the assets.
+            assetDistributions[i] = uint32(riskValues_[i].assetValue * 1_000_000 / totalValue);
```

Comment:
as assetValue is a uint256 in basecurrency, don't think this can overflow


```suggestion
            unchecked {
                assetDistributions[i] = uint32(riskValues_[i].assetValue * 1_000_000 / totalValue);
                ++i;
            }
```